py/build.gradle

```diff
@@ -170,27 +185,24 @@ artifacts {
     sdist distPython
 }
 
-//
-// Test python against Spark cloud
-//
-task testPython(type: Exec, dependsOn: distPython) {
+task testPythonConf(type: Exec, dependsOn: distPython){
     doFirst {
         preparePythonEnv(environment)
     }
-    // Add PySparkling zip on PYTHONPATH so unittests can access all the PySparkling packages
-    environment['PYTHONPATH'] = "${distPython.archivePath}${FPS}${environment['PYTHONPATH']}"
-    environment['spark.ext.h2o.backend.cluster.mode'] = detectBackendClusterMode()
-    environment["spark.ext.h2o.external.start.mode"] = detectExternalBackendStartMode()
-    // sparkling assembly jar is here because of external h2o tests
-    environment["sparkling.assembly.jar"] = configurations.compile.join(',')
-
+    preparePythonTestEnv(environment)
+    commandLine getOsSpecificCommandLine([pythonexec, "tests/tests_unit_conf.py"])
+}
 
+task testPythonConversions(type: Exec, dependsOn: distPython){
+    doFirst {
+        preparePythonEnv(environment)
+    }
+    preparePythonTestEnv(environment)
```

Comment:
Why one preparation is at configuration stage and second one at execution?

Reply:
The `preparePythonTestEnv(environment)` depends on the JAR archive of SW, that's why it is in the execution phase